helpers/pedigree_from_sample_metadata.py

```diff
@@ -0,0 +1,215 @@
+
+        # remove parents and assign an individual sample ID
+        if singles:
+            ped_entry['paternal_id'] = ['0' if plink else '']
+            ped_entry['maternal_id'] = ['0' if plink else '']
+            ped_entry['family_id'] = str(counter)
+
+        else:
+            ped_entry['paternal_id'] = sample_to_cpg_dict.get(
+                ped_entry['paternal_id'], ['0' if plink else '']
+            )
+            ped_entry['maternal_id'] = sample_to_cpg_dict.get(
+                ped_entry['maternal_id'], ['0' if plink else '']
+            )
+
+        new_entries.append(ped_entry)
+
+    return new_entries
+
+
+def write_fat_pedigree(fat_pedigree: List[Dict[str, List[str]]], output: str):
+    """
+    take the pedigree data, and write out as a correctly formatted PED file
+    this permits the situation where we have multiple possible samples per individual
+
+    :param fat_pedigree: the PED content with sample lists
+    :param output: file location to create
+    """
+    with open(output, 'w', encoding='utf-8') as handle:
+        handle.write('\t'.join(PED_KEYS) + '\n')
+        for entry in fat_pedigree:
+            for sample, mother, father in product(
+                entry['individual_id'], entry['paternal_id'], entry['maternal_id']
+            ):
+                handle.write(
+                    '\t'.join(
+                        [
+                            entry['family_id'],
+                            sample,
+                            mother,
+                            father,
+                            str(entry['sex']),
+                            str(entry['affected']),
+                        ]
+                    )
+                    + '\n'
+                )
+
+
+def get_pedigree_for_project(project: str) -> List[Dict[str, str]]:
+    """
+    fetches the project pedigree from sample-metadata
+    list, one dict per participant
+    :param project:
+    :return: all content retrieved from API
+    """
+
+    return FamilyApi().get_pedigree(project=project)
+
+
+def ext_to_int_sample_map(project: str) -> Dict[str, List[str]]:
+    """
+    fetches the participant-sample mapping, so external IDs can be translated
+    to the corresponding CPG ID
+
+    This endpoint returns a list of tuples, each containing a participant ID
+    and corresponding sample ID
+
+    This originally had an expectation of a 1:1 participant:sample relationship
+    that will not hold in general, as numerous samples have multiple samples
+
+    for each participant, create a list of all possible samples
+
+    :param project:
+    :return: the mapping dictionary
+    """
+
+    sample_map = {}
+    for (
+        participant,
+        sample,
+    ) in ParticipantApi().get_external_participant_id_to_internal_sample_id(
+        project=project
+    ):
+        sample_map.setdefault(participant, []).append(sample)
+    return sample_map
```

Comment:
Using a defaultdict is another alternative

```suggestion
    sample_map = defaultdict(list)
    for (
        participant,
        sample,
    ) in ParticipantApi().get_external_participant_id_to_internal_sample_id(
        project=project
    ):
        sample_map[participant].append(sample)
    return sample_map
```